src/edu/csus/ecs/pc2/core/imports/LoadAccounts.java

```diff
@@ -397,4 +534,172 @@ public static void updateAccountsFromLoadAccountsFile(IInternalContest contest,
             contest.storeConfiguration(StaticLog.getLog());
         }
     }
+    
+    
+    
+    /**
+     * Create a list of updated accounts, if account in TSV file does not exist will create a new Account.
+     * 
+     * Intention is a list of all accounts in the TSV files that need to be updated or added.
+     * 
+     * File must have a header of field names.  The fields can appear in any order.
+     * 
+     * Reads TSV file, updates existing accounts in model, returns only a list
+     * of accounts that should be updated.
+     * 
+     * <P>
+     * 
+     * All accounts that are specified in the tsv file must exist, if an account
+     * does not exist then a IllegalTSVFormatException will be thrown.
+     * 
+     * <P>
+     * 
+     * 
+     * 1st line should contain the column headers.   Columns can appear in any order.
+     * Supported column headers are:
+     * <pre>
+     * account (required)
+     * alias
+     * displayname
+     * group (name)
+     * password
+     * permdisplay
+     * permlogin
+     * site (required)
+
+     * </pre>
+     * 
+     * @param filename
+     * @param existingAccounts 
+     * @param groupList
+     * @return an array of accounts
+     * @throws Exception
+     */
+    public Account[] fromTSVFileWithNewAccounts(String filename, Account[] existingAccounts, Group[] groupList) throws Exception  {
+        
+        /**
+         * Output accounts
+         */
+        Map<ClientId, Account> accountMap = new HashMap<ClientId, Account>();
+        
+        if (existingAccounts != null && existingAccounts.length > 0) {
+            for (int i = 0; i < existingAccounts.length; i++) {
+                existingAccountsMap.put(existingAccounts[i].getClientId(), existingAccounts[i]);
+            }
+        }
+        groups.clear();
+        for (Group group : groupList) {
+            groups.put(group.toString(),group);
+        }
+        int lineCount = 0;
+        String[] columns;
+        BufferedReader in = new BufferedReader(new InputStreamReader(new FileInputStream(filename), "UTF8"));
+        String line = in.readLine();
+        while (line != null && line.startsWith("#")) {
+            line = in.readLine();
+            lineCount++;
+        }
+        lineCount++;
+        if (line != null) {
+            columns = TabSeparatedValueParser.parseLine(line);
+            siteColumn = -1;
+            accountColumn = -1;
+            displayNameColumn = -1;
+            passwordColumn = -1;
+            groupColumn = -1;
+            aliasColumn = -1;
+            externalIdColumn = -1;
+            permDisplayColumn = -1;
+            permLoginColumn = -1;
+            permPasswordColumn = -1;
+            scoreAdjustmentColumn = -1;
```

Comment:
The above list of variable initializations fails to include `longSchoolNameColumnn`, `shortSchoolNameColumn`,          `countryCodeColumn`, and `teamNameColumn`.  Those variables indeed have _initialization clauses_ which set them to `-1`, but if someone calls this method _twice_ then the variables need to be (re-)initialized to `-1`.

Note that this same bug exists in the original method (`fromTSVFile(String filename, Account[] existingAccounts, Group[] groupList)`) from which this new method was "cloned".

Reply:
Add initializers in fromTSVFile method and fromTSVFileWithNewAccounts method

Reply:
pushed fix.